### PR TITLE
Add CFMutableDictionary type.

### PR DESCRIPTION
**Pull request by djg**
Added CFMutableDictionary that presents the mutable dictionary
functions of CF. Tries to present a similar interface as CFDictionary.

Fixes issue #93.

---
This change is [Reviewable](https://reviewable.io/reviews/servo/core-foundation-rs/135)

### CODE REVIEW

File: core-foundation/src/dictionary.rs [hunk 1]
```diff
@@ -130,4 +249,133 @@ impl CFDictionary {
...
+        unsafe {
+            CFDictionaryGetTypeID()
+        }
+    }
+}
+
+impl Clone for CFMutableDictionary {
+    #[inline]
+    fn clone(&self) -> CFMutableDictionary {
+        unsafe {
```
Comment (jdm) [inline core-foundation/src/dictionary.rs]: self.copy_with_capacity?

Comment (djg) [reply core-foundation/src/dictionary.rs]: Thanks for the review, @jdm. I pushed the requested changes.

**Comment by jdm** [comment]
`
error: process didn't exit successfully: `/Users/travis/build/servo/core-foundation-rs/core-foundation/target/debug/deps/core_foundation-a1646b9f32dd17a1` (signal: 4, SIGILL: illegal instruction)
`
However, the test that was added in this PR passed, so probably preexisting?

**Comment by jdm** [comment]
I think it's another instance of https://github.com/servo/core-foundation-rs/pull/141 that panics during a C callback.
@bors-servo retry

**Pull request merged**